drivers/virtio/virtqueue.c

```diff
@@ -0,0 +1,186 @@
+/*
```

Comment:
BTW, BTW, can't we use open-amp's virtqueue implementations?

https://github.com/zephyrproject-rtos/open-amp/blob/main/open-amp/lib/virtio/virtqueue.c

Reply:
Initially, we evaluated the use of the OpenAMP library, but we found that it supports only legacy VIRTIO. Meaning, among other things, its API limits us to the first 32 feature bits only, and there are no endianness conversions mandated by the specification.

Reply:
Good call to future-proof it. It would be ideal if OpenAMP could uprev at some point. I know that's a popular framework.

Reply:
@fkokosinski 
Thank you for checking. I fully understand the reason.